corehq/apps/case_search/tests/test_get_related_cases.py

```diff
@@ -81,47 +84,59 @@ def test_get_related_case_results(self):
         self._assert_related_case_ids(cases, {"host", "parent/parent"}, {"c4", "c1"})
 
-    def test_get_related_case_results_duplicates(self):
+    def test_get_related_cases_duplicates(self):
         """Test that `get_related_cases` does not include any cases that are in the initial
         set or are duplicates of others already found."""
 
         # d1 :> c2 > c1 > a1
         # d1 > c1
+        # c2 :> h1
         # Search for case type 'c'
         # - initial results c1, c2
         # - related lookups (parent, parent/parent) yield a1, c1, a1
         # - child lookups yield c2, d1
-        # - (future) extension lookups yield d1
+        # - extension lookups yield d1
+        # - host lookups yield h1
         cases = [
             {'_id': 'a1', 'case_type': 'a'},
+            {'_id': 'h1', 'case_type': 'h'},
             {'_id': 'c1', 'case_type': 'c', 'index': {
                 'parent': ('a', 'a1'),
             }},
             {'_id': 'c2', 'case_type': 'c', 'index': {
                 'parent': ('c', 'c1'),
+                'host': ('h', 'h1', 'extension'),
             }},
             {'_id': 'd1', 'case_type': 'd', 'index': {
                 'parent': ('c', 'c1'),
-                'host': ('c', 'c2'),
+                'host': ('c', 'c2', 'extension'),
             }},
         ]
         self._bootstrap_cases_in_es_for_domain(self.domain, cases)
 
         hits = CaseSearchES().domain(self.domain).case_type("c").run().hits
-        cases = [wrap_case_search_hit(result) for result in hits]
-        self.assertEqual({case.case_id for case in cases}, {'c1', 'c2'})
+        source_cases = [wrap_case_search_hit(result) for result in hits]
+        self.assertEqual({case.case_id for case in source_cases}, {'c1', 'c2'})
 
-        with patch("corehq.apps.case_search.utils.get_related_case_relationships",
+        with patch("corehq.apps.case_search.utils.get_search_detail_relationship_paths",
                    return_value={"parent", "parent/parent"}), \
-             patch("corehq.apps.case_search.utils.get_child_case_types", return_value={"c", "d"}), \
-             patch("corehq.apps.case_search.utils.get_app_cached"):
-            cases = get_related_cases(_QueryHelper(self.domain), None, {"c"}, cases, None)
+                patch("corehq.apps.case_search.utils.get_child_case_types", return_value={"c", "d"}), \
+                patch("corehq.apps.case_search.utils.get_app_cached"):
+            cases = get_related_cases(_QueryHelper(self.domain), None, {"c"}, source_cases, None, False)
 
         case_ids = Counter([case.case_id for case in cases])
         self.assertEqual(set(case_ids), {"a1", "d1"})  # c1, c2 excluded since they are in the initial list
         self.assertEqual(max(case_ids.values()), 1, case_ids)  # no duplicates
 
-    def test_get_related_case_results_expanded_results(self):
+        with patch("corehq.apps.case_search.utils.get_search_detail_relationship_paths",
```

Comment:
Tried to avoid  comments if possible- I separated out calls of `get_related_cases` into a helper function that only takes in the bool `include_all_related_cases` param to make it more obvious. Otherwise, extracted it into a variable.

Can add comments if you think it's still not clear enough
[f1210d9](https://github.com/dimagi/commcare-hq/pull/32587/commits/f1210d96f59b4e13cb74dc4aee78daf78c4f6df0)